Exec/science/flame_wave/problem_setup.H

```diff
@@ -0,0 +1,308 @@
+    model_params.T_star = problem::T_star;
+    model_params.T_hi = problem::T_hi;
+    model_params.T_lo = problem::T_lo;
+
+    model_params.H_star = problem::H_star;
+    model_params.atm_delta = problem::atm_delta;
+
+    model_params.low_density_cutoff = problem::low_density_cutoff;
+
+    generate_initial_model(nx_model + ng,
+                           problo[AMREX_SPACEDIM-1] - ng * problem::dx_model,
+                           probhi[AMREX_SPACEDIM-1],
+                           model_params, 0);
+
+    // now create a perturbed model -- we want the same base conditions
+    // a hotter temperature
+
+    model_params.T_hi = model_params.T_hi + problem::dtemp;
+
+    generate_initial_model(nx_model + ng,
+                           problo[AMREX_SPACEDIM-1] - ng * problem::dx_model,
+                           probhi[AMREX_SPACEDIM-1],
+                           model_params, 1);
+
+    // set center
+
+    for (int d = 0; d < AMREX_SPACEDIM; d++) {
+        problem::center[d] = 0.5_rt * (problo[d] + probhi[d]);
+    }
+
+#if AMREX_SPACEDIM == 2
+    // for axisymmetry, put the x-center on the x-axis
+    // and the y-center at 0, so the height computation is okay
+    problem::center[0] = 0.0_rt;
+    problem::center[1] = 0.0_rt;
+#endif
+
+    // set the ambient state for the upper boundary condition
+
+    ambient::ambient_state[URHO] = model::profile(0).state(model::npts-1, model::idens);
+    ambient::ambient_state[UTEMP] = model::profile(0).state(model::npts-1, model::itemp);
+    for (int n = 0; n < NumSpec; n++) {
+        ambient::ambient_state[UFS+n] =
+            ambient::ambient_state[URHO] * model::profile(0).state(model::npts-1, model::ispec+n);
+    }
+
+    ambient::ambient_state[UMX] = 0.0_rt;
+    ambient::ambient_state[UMY] = 0.0_rt;
+    ambient::ambient_state[UMZ] = 0.0_rt;
+
+    // make the ambient state thermodynamically consistent
+
+    eos_t eos_state;
+    eos_state.rho = ambient::ambient_state[URHO];
+    eos_state.T = ambient::ambient_state[UTEMP];
+    for (int n = 0; n < NumSpec; n++) {
+        eos_state.xn[n] = ambient::ambient_state[UFS+n] / eos_state.rho;
+    }
+
+    eos(eos_input_rt, eos_state);
+
+    ambient::ambient_state[UEINT] = eos_state.rho * eos_state.e;
+    ambient::ambient_state[UEDEN] = eos_state.rho * eos_state.e;
+}
+
+
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+void problem_initialize_state_data (int i, int j, int k,
+                                    Array4<Real> const& state,
+                                    const GeometryData& geomdata)
+{
+
+    const Real* dx = geomdata.CellSize();
+    const Real* problo = geomdata.ProbLo();
+
+    Real x = problo[0] + dx[0] * (static_cast<Real>(i) + 0.5_rt) - problem::center[0];
+
+    Real y = 0.0;
+#if AMREX_SPACEDIM >= 2
+    y = problo[1] + dx[1] * (static_cast<Real>(j) + 0.5_rt) - problem::center[1];
+#endif
+
+    Real z = 0.0;
+#if AMREX_SPACEDIM == 3
+    z = problo[2] + dx[2] * (static_cast<Real>(k) + 0.5_rt);
```

Comment:
What does it mean that `x` and `y` subtract off center but `z` does not?

Reply:
in 2-d, the center is 0, so this is a no op.  In 3-d, we want to be in the geometric center laterally, so we subtract off the center, but for z, we want to know the height from the bottom of the domain for interpolating, so we don't subtract off center.

I suppose I could just set the center in z to be 0 to be consistent.